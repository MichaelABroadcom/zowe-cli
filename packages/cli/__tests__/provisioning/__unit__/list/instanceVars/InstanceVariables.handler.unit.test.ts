/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { ListRegistryInstances, ListInstanceVariables } from "@zowe/provisioning-for-zowe-sdk";

jest.mock("@zowe/provisioning-for-zowe-sdk");
import { IHandlerParameters } from "@zowe/imperative";
import * as Handler from "../../../../../src/provisioning/list/instanceVariables/InstanceVariables.handler";
import { instanceVariables } from "../../../../../src/provisioning/list/instanceVariables/InstanceVariables.definition";
import { ProvisioningListMocks } from "../../../__resources__/ProvisioningListMocks";
import {
    UNIT_TEST_ZOSMF_PROF_OPTS,
    getMockedResponse,
    UNIT_TEST_PROFILES_ZOSMF
} from "../../../../../../../__tests__/__src__/mocks/ZosmfProfileMock";

const DEFAULT_PARAMTERS: IHandlerParameters = {
    arguments: {
        $0: "bright",
        _: ["provisioning", "list", "instance-variables"],
        ...UNIT_TEST_ZOSMF_PROF_OPTS
    },
    positionals: ["provisioning", "list", "instance-variables"],
    response: getMockedResponse(),
    definition: instanceVariables,
    fullDefinition: instanceVariables,
    profiles: UNIT_TEST_PROFILES_ZOSMF
};

describe("list instance variables handler tests", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });
    it("should be able to list instance vars", async () => {
        ListRegistryInstances.listFilteredRegistry = jest.fn(
            (session, zOSMFVersion, instanceId) => {
                return ProvisioningListMocks.LIST_REGISTRY_INSTANCES_RESPONSE;
            }
        );
        ListInstanceVariables.listVariablesCommon = jest.fn(
            (session, zOSMFVersion, instanceId) => {
                return {};
            }
        );
        const handler = new Handler.default();
        const params = Object.assign({}, ...[DEFAULT_PARAMTERS]);
        await handler.process(params);
        expect(ListInstanceVariables.listVariablesCommon).toHaveBeenCalledTimes(
            1
        );
    });
});
