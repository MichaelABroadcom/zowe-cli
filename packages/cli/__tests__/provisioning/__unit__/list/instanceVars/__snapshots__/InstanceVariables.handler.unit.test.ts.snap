// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list instance info handler tests should be able to list instance vars 1`] = `
Object {
  "fields": Array [
    "name",
    "value",
    "visibility",
    "update-registry",
  ],
  "format": "table",
  "header": true,
  "output": undefined,
}
`;

exports[`list instance info handler tests should be able to list instance vars 2`] = `undefined`;
